sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -45,9 +45,13 @@ object CommandUtils extends Logging {
     val catalog = sparkSession.sessionState.catalog
     if (sparkSession.sessionState.conf.autoSizeUpdateEnabled) {
       val newTable = catalog.getTableMetadata(table.identifier)
-      val newSize = CommandUtils.calculateTotalSize(sparkSession, newTable)
-      val newStats = CatalogStatistics(sizeInBytes = newSize)
-      catalog.alterTableStats(table.identifier, Some(newStats))
+      if (newTable.stats == table.stats) {
+        val newSize = CommandUtils.calculateTotalSize(sparkSession, newTable)
+        val newStats = CatalogStatistics(sizeInBytes = newSize)
+        catalog.alterTableStats(table.identifier, Some(newStats))
+      } else {
+        logInfo(s"Skip to update statistics for ${table.qualifiedName} as it updated by Hive.")
```

Comment:
For this, I'm just curious if we can claim this is valid for all (including future) Hive versions (HMS servers and megastore jars `spark.sql.hive.metastore.version`).
> as it updated by Hive.

Reply:
Yes. It supports from 0.13 to 2.3 and 3.1.
Only Hive 0.12 does not calculate statistics. So it will be calculated by Spark.